logic/teamLogic.js

```diff
@@ -0,0 +1,37 @@
+const MongoClient = require('mongodb').MongoClient
+const MsLogger = require('@first-lego-league/ms-logger').Logger()
+
+const MONGU_URI = process.env.MONGO
+
+module.exports.deleteValidation = function (params) {
+  try {
+    deleteMatchesForTeam(parseInt(params.id))
+    return true
+  } catch (e) {
+    return false
+  }
+}
+
+function deleteMatchesForTeam (teamNumber) {
+  MongoClient.connect(MONGU_URI).then(connection => {
+    connection.db().collection('matches').find({ 'matchTeams.teamNumber': teamNumber }).toArray().then(data => {
+      for (const match of data) {
+        for (const matchTeam of match.matchTeams) {
```

Comment:
This double `for` loop is needed? I thought the query in the `find` methods does this filtering for you...

Reply:
The find returns the whole document, not just the internal document we want to edit

Reply:
You can use [projection](https://docs.mongodb.com/manual/reference/operator/projection/)